pkg/authentication/tokens.go

```diff
@@ -0,0 +1,64 @@
+package authentication
+
+import (
+	"context"
+	"errors"
+	"github.com/golang-jwt/jwt/v5"
+	"strings"
+)
+
+// auth0 is an Authentication implementation using Auth0 as an authentication provider.
+type auth0 struct {
+	publicKey []byte
+}
+
+// VerifyJWT verifies that the given token is a valid JWT and was correctly signed by Auth0.
+func (auth *auth0) VerifyJWT(ctx context.Context, token string) error {
+	if err := auth.validateToken(token); err != nil {
+		return err
+	}
+	parsedToken, err := jwt.Parse(token, auth.keyFunc)
+	if err != nil {
+		return err
+	}
+	if !parsedToken.Valid {
+		return errors.New("token is not valid")
+	}
+	return nil
+}
+
+// validateToken validates that the given token is a valid JWT.
+func (auth *auth0) validateToken(token string) error {
```

Comment:
nit: Since we might have to support other types of tokens (e.g. private tokens), let's play it safe and rename this to `validateJWT`.